feat: 添加拖拽插入、酒馆预设排序、文件导入导出、国际化支持、卡片折叠展开等多项改进

samiytaa opened the pull request:
## STPresetEditor 新增功能

### 🎯 交互体验优化
- **按钮切换**: 左侧边栏已添加状态的按钮可再次点击回到未添加状态
- **拖拽插入**: 支持从左侧Prompt Library拖拽卡片到中间编辑器区域
- **卡片折叠/展开**: 编辑器中的卡片支持折叠/展开，提供一键折叠/展开所有卡片功能

### 📁 文件操作增强
- **JSON文件导入**: 支持直接导入JSON文件，提供文件选择和拖拽上传两种方式
- **导出功能优化**: 支持导出JSON文件下载，文件名可自定义并自动添加日期后缀
- **SillyTavern排序**: 完全兼容SillyTavern的预设格式和排序机制

### 🔍 搜索功能
- **Editor区域搜索**: 在编辑器区域添加搜索功能，与Prompt Library风格保持一致
- **实时过滤**: 支持按名称和ID进行模糊匹配，使用防抖优化性能

### 🌐 国际化支持
- **多语言切换**: 支持英文和简体中文切换，提供圆形语言切换按钮
- **完整覆盖**: 所有界面文本、按钮、提示信息都支持国际化
- **持久化存储**: 用户语言偏好自动保存

## 使用方式
1. 双击 `启动STPresetEditor.bat` 启动应用
2. 支持导入JSON文件或直接编辑
3. 使用拖拽、搜索、折叠等功能提升编辑效率
4. 导出编辑完成的预设文件

Commit:
feat: 添加国际化支持、文件导入导出、搜索功能等多项改进

- 添加中英文国际化支持，支持语言切换
- 支持JSON文件导入导出功能
- 添加编辑器搜索功能，支持按名称和ID过滤
- 优化拖拽插入和卡片折叠功能
- 改进UI/UX体验，使用Headless UI组件
- 添加变量管理和宏分析功能
- 支持实时编辑和自动保存
- 完善响应式设计，支持移动端

Commit:
feat: 添加拖拽排序、国际化支持、文件导入导出等多项改进

- 优化拖拽插入和卡片折叠功能，支持可视化排序
- 添加中英文国际化支持，支持语言切换
- 支持JSON文件导入导出功能
- 添加编辑器搜索功能，支持按名称和ID过滤
- 改进UI/UX体验，使用Headless UI组件
- 添加变量管理和宏分析功能
- 支持实时编辑和自动保存
- 完善响应式设计，支持移动端

Commit:
docs: 更新README.md为中文功能总结

- 用功能总结内容替换原有README
- 突出显示拖拽排序、国际化、文件操作等核心功能
- 提供清晰的中文使用说明

Commit:
feat: 添加STPresetEditor批处理脚本

- 新增STPresetEditor.bat文件，用于自动化项目环境设置
- 检查package.json文件是否存在，并在缺失时提示用户
- 自动安装node_modules依赖，确保项目可以正常运行
- 提供开发服务器启动指令，并提示用户如何停止服务

samiytaa committed:
Update README.md

samiytaa committed:
Update README.md

samiytaa committed:
启动STPresetEditor.bat

Nativu5 commented on src/stores/presetStore.js:
```diff
@@ -29,6 +30,7 @@ function debouncedAction(action, delay) {
 export const usePresetStore = defineStore('preset', {
   state: () => ({
     rawJson: '',
+    originalFilename: '', // 存储导入的原始文件名
```
Please write comment in English.

Nativu5 commented on README.md:
Please do not overwrite README.md of the whole project. New features could be stated within the existing README framework. 

In addition, I strongly suggest adding a part in GEMINI.md to reflect the code changes, which makes Code Agents understand the codebase more easily.

Nativu5 commented:
Thanks again for your contribution. As soon as the comments are addressed, we can bring the commits into mainline.

samiytaa committed:
更新 9.15

更新 9.15

samiytaa commented:
> Thanks again for your contribution. As soon as the comments are addressed, we can bring the commits into mainline.

(#^.^#) I modified the code as required, made some adjustments, and added the word replacement/prefix-suffix functionality.

samiytaa commented:
I'd like to ask, hoping it doesn't offend : I saw this project on Discord before, but that post is no longer visible. Are you still planning to share it on Discord? 😊 If you're not posting, would it be okay for me to share a new version? 🙌

samiytaa committed:
9.22 修复部分逻辑错误

samiytaa committed:
Delete 启动STPresetEditor.bat

samiytaa committed:
Merge branch 'master' of https://github.com/samiytaa/STPresetEditor-s

Nativu5 commented:
> I'd like to ask, hoping it doesn't offend : I saw this project on Discord before, but that post is no longer visible. Are you still planning to share it on Discord? 😊 If you're not posting, would it be okay for me to share a new version? 🙌

Sorry, just had a busy week. The post on discord is still there but no one is sending messages. I'll merge these as soon as possible, after I check if this will break the cloudflare deployment.
